packages/common/src/transports/wiresaw.ts

```diff
@@ -0,0 +1,163 @@
+import { estimateUserOperationGas } from "./methods/estimateUserOperationGas";
+import { getUserOperationReceipt } from "./methods/getUserOperationReceipt";
+
+type WiresawSendUserOperationResult = {
+  txHash: Hex;
+  userOpHash: Hex;
+};
+
+type WiresawOptions<transport extends Transport> = {
+  wiresaw: transport;
+  fallbackBundler?: Transport;
+  fallbackEth?: Transport;
+};
+
+export function wiresaw<const wiresawTransport extends Transport>(
+  transport: WiresawOptions<wiresawTransport>,
+): wiresawTransport {
+  return ((opts) => {
+    const { request: originalRequest, ...rest } = transport.wiresaw(opts);
+
+    let chainId: Hex | null = null;
+    const transactionHashes: { [userOpHash: Hex]: Hex } = {};
+    const transactionReceipts: { [transactionHashes: Hex]: RpcTransactionReceipt } = {};
+
+    return {
+      ...rest,
+      // TODO: type `request` so we don't have to cast
+      async request(req): ReturnType<EIP1193RequestFn> {
+        try {
+          if (req.method === "eth_chainId") {
+            if (chainId != null) return chainId;
+            if (transport.fallbackEth) {
+              const { request: fallbackRequest } = transport.fallbackEth(opts);
+              return (chainId = await fallbackRequest(req));
+            }
+            return (chainId = await originalRequest(req));
+          }
+
+          if (req.method === "eth_estimateGas") {
+            return await originalRequest({ ...req, method: "wiresaw_estimateGas" });
+          }
+
+          if (req.method === "eth_call") {
+            return await originalRequest({ ...req, method: "wiresaw_call" });
+          }
+
+          if (req.method === "eth_getTransactionCount") {
+            return await originalRequest({ ...req, method: "wiresaw_getTransactionCount" });
+          }
+
+          if (req.method === "eth_getTransactionReceipt") {
+            return await getTransactionReceipt((req.params as [Hex])[0]);
+          }
+
+          if (req.method === "eth_sendUserOperation") {
+            const { userOpHash, txHash } = (await originalRequest({
+              ...req,
+              method: "wiresaw_sendUserOperation",
+            })) as WiresawSendUserOperationResult;
+            transactionHashes[userOpHash] = txHash;
+            return userOpHash;
+          }
+
+          if (req.method === "eth_getUserOperationReceipt") {
+            const userOpHash = (req.params as [Hex])[0];
+            const knownTransactionHash = transactionHashes[userOpHash];
+            if (knownTransactionHash) {
+              const transactionReceipt = await getTransactionReceipt(knownTransactionHash);
+              if (transactionReceipt) {
+                return getUserOperationReceipt(userOpHash, transactionReceipt);
+              }
+            }
+            if (transport.fallbackBundler) {
+              const { request: fallbackRequest } = transport.fallbackBundler(opts);
+              return await fallbackRequest(req);
+            }
```

Comment:
we don't ask wiresaw because it's not supported? should we throw here and let the `catch` fallback below handle this?

Reply:
yeah this method is not supported on wiresaw, but we can still do it via `getTransactionReceipt`, but only for known transaction hashes (those where we previously sent the user operation through this transport). And yep, can let the catch handle it.